File: Second Year (2013-2014)/OMGL7/TP3 - Website in JSP/src/db/CommandeDAO.java
package db;

import java.sql.*;
/**
* @author Andriamilanto Tompoariniaina - Canno Camille - Info2B
*/

/**
* Classe pour les objets DAO de Commande
*/
public class CommandeDAO {
//Methodes
	/**
	*Constructeur
	*/
	public CommandeDAO() {}

	/**
	*Pour retrouver tous les tuples
	*/
	public ResultSet findAll() {
		Statement stat = null;
		String query = "";
		ResultSet ret = null;

		try {
			//Recuperation de la connexion
			Connection con = MySQLConnection.getInstance().getConnection();

			//Preparation de la requete en ligne
			stat = con.createStatement();

			//Preparation de la requete
			query = "SELECT * FROM Commande;";

			//Le resultat a retourner
			ret = stat.executeQuery(query);
		}
		catch (SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}

		//Retourne l'execution de la requete sous la forme d'un objet ResultSet
		return ret;
	}

	/**
	*Permet de retrouver juste un tuple
	*@param id idCommande de la commande a retrouver
	*/
	public Commande find(int numCommande) {
		Statement stat = null;
		String query = "";
		Commande ret = new Commande();

		try {
			//Recuperation de la connexion
			Connection con = MySQLConnection.getInstance().getConnection();

			//Preparation de la requete en ligne
			stat = con.createStatement();

			//Preparation de la requete
			query = "SELECT * FROM Commande WHERE NumeroCommande = " + numCommande + ";";

			//Retourne l'execution de la requete sous la forme d'un objet ResultSet
			ResultSet result = stat.executeQuery(query);

			//Si le resultat est bon, prends la premiere ligne
			if (result.first()) {
				ret.init(numCommande, result.getString(2), result.getString(3), result.getInt(4));
			}
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}

		return ret;
	}


	/**
    *Methode qui permet d'inserer un tuple
    *@param tuple Objet de type Commande a inserer
    */
    public void insert(Commande tuple) {
    	Statement stat = null;
        String query = "";

        //Recuperation des attributs de l'objet Commande
        int numCommande = tuple.getNumeroCommande();
        String clientCommande = tuple.getClientCommande();
        String titreCommande = tuple.getTitreCommande();
        int total = tuple.getTotal();

        try {
	    	//Recuperation de la connexion
	        Connection con = MySQLConnection.getInstance().getConnection();

	        //Preparation de la requete
	        query = "INSERT INTO Commande VALUES(" + numCommande + "," + clientCommande + "," + titreCommande + "," + total + ");";

	        //Execute la requête
			stat.executeQuery(query);
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}
    }


    /**
    * Permet de faire la mise a jour d'un tuple
    *@param tuple Objet nouveau a mettre a la place de l'ancien
    */
    public void update(Commande tuple) {
    	Statement stat = null;
        String query = "";

        //Recuperation des attributs de l'objet Commande
        int numCommande = tuple.getNumeroCommande();
        String clientCommande = tuple.getClientCommande();
        String titreCommande = tuple.getTitreCommande();
        int total = tuple.getTotal();

        try {
	    	//Recuperation de la connexion
	        Connection con = MySQLConnection.getInstance().getConnection();

	        //Preparation de la requete
	        query = "UPDATE Commande SET ClientCommande = " + clientCommande + ", TitreCommande = " + titreCommande + ", Total = " + total + " WHERE NumeroCommande = " + numCommande + ";";

	        //Execute la requête
			stat.executeQuery(query);
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}
    }


    /**
    * Permet de supprimer un tuple
    *@param numCommande id du tuple a supprimer
    */
    public void delete(int numCommande) {
    	Statement stat = null;
        String query = "";

        try {
	    	//Recuperation de la connexion
	        Connection con = MySQLConnection.getInstance().getConnection();

	        //Preparation de la requete
	        query = "DELETE FROM Commande WHERE NumeroCommande = " + numCommande + ";";

	        //Execute la requête
			stat.executeQuery(query);
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}
    }
}

File: Second Year (2013-2014)/OMGL7/TP2 - Website in PHP/musiconline/musiconline/db/CommandeDao.php
<?php
/* Authors: ANDRIAMILANTO Tompoariniaina && CANNO Camille
* Groupe: B1
* File: CommandeDao.php
*/

import ('phpens.framework.Application');
import ('phpens.framework.db.DataObject');
import ('phpens.framework.db.DataAccessObject');
import ('musiconline.db.Commande');


class CommandeDao implements DataAccessObject{
  //Attribut instance de DataAccessObject
  private static $dao;

  //Constructeur vide
  private function __construct() { }

  //Permet de retourner une instance de CommandeDAO
  public final static function getInstance() {

    //Vérifie si l'objet est correcte
    if(!isset(self::$dao)) {

      //Sinon, il en créer un vide
      self::$dao= new CommandeDao();
    }

    //Retourne l'objet ou l'objet créer
    return self::$dao;
  }


  //Retourne tous les tuples
  public final function findAll(){

    //Récupère la méthode de connexion au SGBD
    $dbc = Application::getInstance()->getDBConnection();

    //Prépare la requête
    $query = "select * from Commande";

    //Prépare la requête en dur
    $stmt = $dbc->query($query);

    //Enregistre tous les tuples dans un tableau d'instances de classe
    $results = $stmt->fetchALL(PDO::FETCH_CLASS, 'Commande');

    //Retourne le résultat
    return $results;
   }


  //Permet d'insérer un tuple
  public final function insert(DataObject $do){

    //Vérifie si l'objet entré est instance de commande
    if($do instanceof Commande){

      //Récupère la méthode de connexion
      $dbc = Application::getInstance()->getDBConnection();

      //Prépare la requête
      $query = "insert into Commande(NumeroCommande,ClientCommande,TitreCommande,Total) values (:NumeroCommande,:ClientCommande,:TitreCommande,:Total)";

      //Prépare la requête en dur
      $stmt = $dbc->prepare($query);

      //Remplace les paramètres
      $stmt->bindValue(':NumeroCommande',$do->getNUmeroCommande(),PDO::PARAM_STR);
      $stmt->bindValue(':ClientCommande',$do->getCLientCommande(),PDO::PARAM_STR);
      $stmt->bindValue(':TitreCommande',$do->getTItreCommande(),PDO::PARAM_STR);
      $stmt->bindValue(':Total',$do->getTOtal(),PDO::PARAM_STR);

      //Exécute la requête
      $stmt->execute($data);
    }
 }


  //Permet de mettre à jour un tuple
  public function update(DataObject $obj){

	  //Vérifie si l'objet entré est instance de commande
    if($obj instanceof Commande){

      //Récupère la méthode de connexion
      $dbc = Application::getInstance()->getDBConnection();

      //Prépare la requête
      $query = "UPDATE Commande SET TitreCommande = :TitreCommande, Total = :Total WHERE NumeroCommande = :NumeroCommande AND ClientCommande = :ClientCommande";

      //Prépare la requête en dur
      $stmt = $dbc->prepare($query);

      //Remplace les paramètres
      $stmt->bindValue(':NumeroCommande',$obj->getNUmeroCommande(),PDO::PARAM_STR);
      $stmt->bindValue(':ClientCommande',$obj->getCLientCommande(),PDO::PARAM_STR);
      $stmt->bindValue(':TitreCommande',$obj->getTItreCommande(),PDO::PARAM_STR);
      $stmt->bindValue(':Total',$obj->getTOtal(),PDO::PARAM_STR);

      //Exécute la requête
      $stmt->execute($data);
  }

  public function delete(DataObject $obj){

    //Vérifie si l'objet entré est instance de commande
    if($obj instanceof Commande){

      //Récupère la méthode de connexion
      $dbc = Application::getInstance()->getDBConnection();

      //Prépare la requête
      $query = "DELETE FROM Commande WHERE NumeroCommande = :NumeroCommande AND ClientCommande = :ClientCommande";

      //Prépare la requête en dur
      $stmt = $dbc->prepare($query);

      //Remplace les paramètres
      $stmt->bindValue(':NumeroCommande',$obj->getNUmeroCommande(),PDO::PARAM_STR);
      $stmt->bindValue(':ClientCommande',$obj->getCLientCommande(),PDO::PARAM_STR);

      //Exécute la requête
      $stmt->execute($data);
  }
}
?>
